src/main/kotlin/com/github/kawamataryo/copygitlink/CopyAsMarkdownWithCode.kt

```diff
@@ -0,0 +1,56 @@
+package com.github.kawamataryo.copygitlink
+
+import GitLink
+import com.intellij.notification.NotificationType
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.CommonDataKeys
+import com.intellij.openapi.actionSystem.PlatformDataKeys
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.PsiFile
+import copyToClipboard
+import showNotification
+import truncateText
+
+
+class CopyAsMarkdownWithCode : AnAction() {
+    override fun actionPerformed(e: AnActionEvent) {
+        val gitLink = GitLink(e)
+
+        if (gitLink.hasRepository) {
+            val permalink = gitLink.permalink
+            val linkText = gitLink.relativePath + gitLink.linePath
+            val markdownLink = "[$linkText]($permalink)"
+            val source = gitLink.source;
+            val editor: Editor =
+                e.getRequiredData(CommonDataKeys.EDITOR)
+
+            var vf:VirtualFile = e.getRequiredData(CommonDataKeys.VIRTUAL_FILE)
+
+
+            val content = """$markdownLink
+                |```${vf.name.substring(vf.name.lastIndex)}
```

Comment:
This is not working in my environment.
If you want to get the extension.

You should write it this way.

```
${vf.name.split(".").last()}
```